e2e/e2e_test.go

```diff
@@ -0,0 +1,135 @@
+package e2e
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"testing"
+
+	"github.com/spectralops/teller/e2e/register"
+	_ "github.com/spectralops/teller/e2e/tests"
+	"github.com/spectralops/teller/e2e/testutils"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	replaceToStaticPath     = "DYNAMIC-FULL-PATH"
+	removeBinaryPlaceholder = "<binary-path>"
+	testsFolder             = "tests"
+)
+
+func TestE2E(t *testing.T) {
+	t.Parallel()
+
+	// validate given binary path
+	binaryPath, err := getBinaryPath()
+	if err != nil {
+		assert.FailNow(t, err.Error())
+	}
+
+	snapshotSuites, err := testutils.GetYmlSnapshotSuites(testsFolder)
+	assert.Nil(t, err)
+
+	// Loop on all test/*.yml files
+	for _, snapshot := range snapshotSuites {
+
+		t.Run(snapshot.Name, func(t *testing.T) {
+
+			// create a temp folder for the test
+			tempFolder, err := os.MkdirTemp(t.TempDir(), strings.ReplaceAll(snapshot.Name, " ", ""))
+			// descrive the base snapshot data of the test
+			snapshotFolder := filepath.Join(tempFolder, "snapshot")
+			assert.Nil(t, err, "could not create temp folder")
+			defer os.RemoveAll(tempFolder)
+
+			if len(snapshot.InitSnapshot) > 0 {
+				err = snapshot.CreateSnapshotData(snapshot.InitSnapshot, snapshotFolder)
+				assert.Nil(t, err)
+			}
+
+			if snapshot.ConfigFileName != "" {
+				err = snapshot.CrateConfig(snapshotFolder)
+				assert.Nil(t, err)
+			}
+
+			flagsCommand := strings.TrimPrefix(snapshot.Command, removeBinaryPlaceholder)
+			stdout, stderr, err := testutils.ExecCmd(binaryPath, strings.Split(flagsCommand, " "), snapshotFolder)
+			assert.Nil(t, err, stderr)
+
+			// In case the stdout/stderr include the dynamic folder path, we want to replace with static-content for better snapshot text compare
+			stdout, stderr = replaceFolderName(stdout, stderr, snapshotFolder)
+
+			if snapshot.ExpectedStdOut != "" {
+				assert.Equal(t, snapshot.ExpectedStdOut, stdout)
+			}
+
+			if snapshot.ExpectedStdErr != "" {
+				assert.Equal(t, snapshot.ExpectedStdErr, stderr)
+			}
+
+			if len(snapshot.ExpectedSnapshot) > 0 {
+				destSnapshotFolder := filepath.Join(tempFolder, "dest")
+				err = snapshot.CreateSnapshotData(snapshot.ExpectedSnapshot, destSnapshotFolder)
+				assert.Nil(t, err)
+
+				// consider to replace `diff` command which is depended on OS to golang plugin.
+				// could't find something better
+				s, _, err := testutils.ExecCmd("diff", []string{"-qr", "-x", snapshot.ConfigFileName, destSnapshotFolder, snapshotFolder}, "")
```

Comment:
might want to encapsulate that under a method/interface that has a clear input/output.
that way when we come to find a different solution - we only need to implement to an interface
interface Differ{
   Diff(...)
}

Reply:
totally agree, done